chain-signatures/contract/tests/user_views.rs

```diff
@@ -0,0 +1,99 @@
+pub mod common;
+use common::{create_response, init_env};
+
+use mpc_contract::primitives::SignRequest;
+
+use near_sdk::{CurveType, NearToken, PublicKey};
+use serde_json::json;
+use std::str::FromStr;
+#[tokio::test]
+async fn test_key_version() -> anyhow::Result<()> {
+    let (_, contract, _, _) = init_env().await;
+
+    let version: u32 = contract
+        .view("latest_key_version")
+        .await
+        .unwrap()
+        .json()
+        .unwrap();
+    assert_eq!(version, 0);
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_public_key() -> anyhow::Result<()> {
+    let (_, contract, _, _) = init_env().await;
+
+    let key: String = contract.view("public_key").await.unwrap().json().unwrap();
+    println!("{:?}", key);
+    let pk = PublicKey::from_str(&key)?;
+    assert_eq!(pk.curve_type(), CurveType::SECP256K1);
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_derived_public_key() -> anyhow::Result<()> {
```

Comment:
We will need to execute actual derivation logic in such a test. Not a blocker now.

Reply:
agree. Probably good to be able to fix the pk in init_env() so we can test the value of the derived pk is correct.